lesson 6/Tensor/app/src/main/cpp/native-lib.cpp

```diff
@@ -0,0 +1,95 @@
+#include <jni.h>
+#include <string>
+#include <string.h>
+#include <sys/socket.h>	//socket
+#include <arpa/inet.h>	//inet_addr
+#include <unistd.h>
+#include <android/log.h>
+#include <errno.h>
+#include <locale>
+
+
+#define LOG_TAG "CLIENT_APP"
+
+jbyteArray getImage(jobject context, JNIEnv* env) {
```

Comment:
And in general, for code readability.
You should structure it, like all `jclass`s should be structured together, all `jmethodID`s together and so on. So please check my implementation and try to get the idea of clear JNI code implementation